src/dog.sol

```diff
@@ -176,6 +180,7 @@ contract Dog is LibNote {
             id = ClipperLike(milk.clip).kick({
                 tab: tab,
                 lot: dink,
+                lip: div(mul(art, rate), ink),
```

Comment:
I think in principle we could save some gas here by doing:
```
due / dink
```
`dink` is required to be non-zero above so we don't need the safe division, and `due / dink` should have approximately the same value as `art * rate / ink`. I don't see any additional precision concerns there under normal risk parameter conditions, given the `dust` checks.

Reply:
Added 👍